ambry-clustermap/src/main/java/com/github/ambry/clustermap/CloudServiceClusterChangeHandler.java

```diff
@@ -1,157 +0,0 @@
-/*
- * Copyright 2020 LinkedIn Corp. All rights reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- */
-package com.github.ambry.clustermap;
-
-import com.github.ambry.clustermap.HelixClusterManager.ClusterChangeHandlerCallback;
-import com.github.ambry.config.ClusterMapConfig;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.stream.Stream;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-
-/**
- * Currently, this just adds a single {@link ReplicaType#CLOUD_BACKED} replica every time a partition is added to allow
- * for requests to be routed to cloud services. Currently it is only registered as a {@link ClusterMapChangeListener},
- * but once the clustermap natively supports VCR clusters, this class can implement {@link HelixClusterChangeHandler}
- * and listen for changes in the VCR cluster.
- */
-class CloudServiceClusterChangeHandler implements ClusterMapChangeListener, ClusterChangeHandler {
-  private static final Logger logger = LoggerFactory.getLogger(CloudServiceClusterChangeHandler.class);
-  private final Set<String> knownPartitions = ConcurrentHashMap.newKeySet();
-  private final ConcurrentHashMap<String, CloudServiceReplica> partitionToReplica = new ConcurrentHashMap<>();
-  private final List<ClusterMapChangeListener> listeners = new CopyOnWriteArrayList<>();
-  private final AtomicLong errorCount = new AtomicLong(0);
-  private final String dcName;
-  private final ClusterMapConfig clusterMapConfig;
-  private final ClusterChangeHandlerCallback changeHandlerCallback;
-  private final CloudServiceDataNode cloudServiceDataNode;
-
-  public CloudServiceClusterChangeHandler(String dcName, ClusterMapConfig clusterMapConfig,
-      ClusterChangeHandlerCallback changeHandlerCallback) throws Exception {
-    this.dcName = dcName;
-    this.clusterMapConfig = clusterMapConfig;
-    this.changeHandlerCallback = changeHandlerCallback;
-    cloudServiceDataNode = new CloudServiceDataNode(dcName, clusterMapConfig);
```

Comment:
Since we no longer support cloud-based data nodes, does this mean we can remove CloudServiceDataNode too?  Please make sure this change will not affect VCR.

Reply:
Yeah, we can remove CloudServiceDataNode too. I can try to put up a separate PR for remove rest of code related to cloud-based data nodes. 

Yeah, this shouldn't affect VCR. I think CloudServiceClusterChangeHandler was added to support cloud-based replicas in cluster map.